sound/soc/sof/intel/hda.c

```diff
@@ -94,9 +90,7 @@ void hda_dsp_block_write(struct snd_sof_dev *sdev, u32 offset, void *src,
 		 */
 		__ioread32_copy(&tmp, dest + m * 4, 1);
 
-		dst_byte = (u8 *)&tmp;
-		for (i = 0; i < n; i++)
-			dst_byte[i] = src_byte[m * 4 + i];
```

Comment:
I honestly prefer this array notation over all the rest. I don't see the point of complex formulas that aren't readable. Let the compiler work for you...

Reply:
@plbossart This is actually from Andy Shevchenko's comment. @keyonjie takes sick leave today. I will revert this part if you prefer the original one.
>> +	if (n) {
>> +		for (i = 0; i < n; i++)
>> +			tmp |= (u32)*(src_byte + m * 4 + i) << (i * 8);
> 
> I suppose src_byte is aligned on 32-bit.
> 
> Why not just
> 
> tmp = *(u32 *)src_byte & ((1 << (8 * n)) - 1);
> 
> 1 -> 0xff
> 2 -> 0xffff
> 3 -> 0xffffff

Reply:
Can you just use the array solution everywhere? It's not like this pointer arithmetic and masks gain us anything.

Reply:
@plbossart this was what Andy suggested and I think it is good as it remove usage of for loop, and it actually improve efficiency?

Reply:
Sorry @keyonjie I think we have to consider about @plbossart 's opinion. So, I made a PR which contains the readq/writeq change only.

Reply:
Ah, I missed the fact that the loop was gone. So yes indeed it's simpler without a loop, thanks  for pushing back. Please add a new PR to improve this (I'd like something more readable though)